File: decompiled/sources/com/hjq/shape/builder/TextColorBuilder.java
package com.hjq.shape.builder;

import android.content.res.ColorStateList;
import android.content.res.TypedArray;
import android.text.SpannableStringBuilder;
import android.widget.TextView;
import com.hjq.shape.other.LinearGradientFontSpan;
import com.hjq.shape.styleable.ITextColorStyleable;

public final class TextColorBuilder {
    private Integer mTextCheckedColor;
    private int mTextColor;
    private Integer mTextDisabledColor;
    private Integer mTextFocusedColor;
    private int[] mTextGradientColor;
    private int mTextGradientOrientation;
    private Integer mTextPressedColor;
    private Integer mTextSelectedColor;
    private final TextView mTextView;

    public TextColorBuilder(TextView textView, TypedArray typedArray, ITextColorStyleable iTextColorStyleable) {
        this.mTextView = textView;
        this.mTextColor = typedArray.getColor(iTextColorStyleable.getTextColorStyleable(), textView.getTextColors().getDefaultColor());
        if (typedArray.hasValue(iTextColorStyleable.getTextPressedColorStyleable())) {
            this.mTextPressedColor = Integer.valueOf(typedArray.getColor(iTextColorStyleable.getTextPressedColorStyleable(), this.mTextColor));
        }
        if (iTextColorStyleable.getTextCheckedColorStyleable() > 0 && typedArray.hasValue(iTextColorStyleable.getTextCheckedColorStyleable())) {
            this.mTextCheckedColor = Integer.valueOf(typedArray.getColor(iTextColorStyleable.getTextCheckedColorStyleable(), this.mTextColor));
        }
        if (typedArray.hasValue(iTextColorStyleable.getTextDisabledColorStyleable())) {
            this.mTextDisabledColor = Integer.valueOf(typedArray.getColor(iTextColorStyleable.getTextDisabledColorStyleable(), this.mTextColor));
        }
        if (typedArray.hasValue(iTextColorStyleable.getTextFocusedColorStyleable())) {
            this.mTextFocusedColor = Integer.valueOf(typedArray.getColor(iTextColorStyleable.getTextFocusedColorStyleable(), this.mTextColor));
        }
        if (typedArray.hasValue(iTextColorStyleable.getTextSelectedColorStyleable())) {
            this.mTextSelectedColor = Integer.valueOf(typedArray.getColor(iTextColorStyleable.getTextSelectedColorStyleable(), this.mTextColor));
        }
        if (typedArray.hasValue(iTextColorStyleable.getTextStartColorStyleable()) && typedArray.hasValue(iTextColorStyleable.getTextEndColorStyleable())) {
            if (typedArray.hasValue(iTextColorStyleable.getTextCenterColorStyleable())) {
                this.mTextGradientColor = new int[]{typedArray.getColor(iTextColorStyleable.getTextStartColorStyleable(), this.mTextColor), typedArray.getColor(iTextColorStyleable.getTextCenterColorStyleable(), this.mTextColor), typedArray.getColor(iTextColorStyleable.getTextEndColorStyleable(), this.mTextColor)};
            } else {
                this.mTextGradientColor = new int[]{typedArray.getColor(iTextColorStyleable.getTextStartColorStyleable(), this.mTextColor), typedArray.getColor(iTextColorStyleable.getTextEndColorStyleable(), this.mTextColor)};
            }
        }
        this.mTextGradientOrientation = typedArray.getColor(iTextColorStyleable.getTextGradientOrientationStyleable(), 0);
    }

    public TextColorBuilder setTextColor(Integer num) {
        this.mTextColor = num.intValue();
        clearTextGradientColor();
        return this;
    }

    public Integer getTextColor() {
        return Integer.valueOf(this.mTextColor);
    }

    public TextColorBuilder setTextPressedColor(Integer num) {
        this.mTextPressedColor = num;
        return this;
    }

    public Integer getTextPressedColor() {
        return this.mTextPressedColor;
    }

    public TextColorBuilder setTextCheckedColor(Integer num) {
        this.mTextCheckedColor = num;
        return this;
    }

    public Integer getTextCheckedColor() {
        return this.mTextCheckedColor;
    }

    public TextColorBuilder setTextDisabledColor(Integer num) {
        this.mTextDisabledColor = num;
        return this;
    }

    public Integer getTextDisabledColor() {
        return this.mTextDisabledColor;
    }

    public TextColorBuilder setTextFocusedColor(Integer num) {
        this.mTextFocusedColor = num;
        return this;
    }

    public Integer getTextFocusedColor() {
        return this.mTextFocusedColor;
    }

    public TextColorBuilder setTextSelectedColor(Integer num) {
        this.mTextSelectedColor = num;
        return this;
    }

    public Integer getTextSelectedColor() {
        return this.mTextSelectedColor;
    }

    public TextColorBuilder setTextGradientColor(int[] iArr) {
        this.mTextGradientColor = iArr;
        return this;
    }

    public int[] getTextGradientColor() {
        return this.mTextGradientColor;
    }

    public TextColorBuilder setTextGradientOrientation(int i) {
        this.mTextGradientOrientation = i;
        return this;
    }

    public Integer getTextGradientOrientation() {
        return Integer.valueOf(this.mTextGradientOrientation);
    }

    public boolean isTextGradientColor() {
        int[] iArr = this.mTextGradientColor;
        return iArr != null && iArr.length > 0;
    }

    public void clearTextGradientColor() {
        this.mTextGradientColor = null;
    }

    public SpannableStringBuilder buildLinearGradientSpannable(CharSequence charSequence) {
        return LinearGradientFontSpan.buildLinearGradientSpannable(charSequence, this.mTextGradientColor, (float[]) null, this.mTextGradientOrientation);
    }

    public ColorStateList buildColorState() {
        int i;
        Integer num = this.mTextPressedColor;
        if (num == null && this.mTextCheckedColor == null && this.mTextDisabledColor == null && this.mTextFocusedColor == null && this.mTextSelectedColor == null) {
            return ColorStateList.valueOf(this.mTextColor);
        }
        int[][] iArr = new int[6][];
        int[] iArr2 = new int[6];
        if (num != null) {
            iArr[0] = new int[]{16842919};
            iArr2[0] = num.intValue();
            i = 1;
        } else {
            i = 0;
        }
        Integer num2 = this.mTextCheckedColor;
        if (num2 != null) {
            iArr[i] = new int[]{16842912};
            iArr2[i] = num2.intValue();
            i++;
        }
        Integer num3 = this.mTextDisabledColor;
        if (num3 != null) {
            iArr[i] = new int[]{-16842910};
            iArr2[i] = num3.intValue();
            i++;
        }
        Integer num4 = this.mTextFocusedColor;
        if (num4 != null) {
            iArr[i] = new int[]{16842908};
            iArr2[i] = num4.intValue();
            i++;
        }
        Integer num5 = this.mTextSelectedColor;
        if (num5 != null) {
            iArr[i] = new int[]{16842913};
            iArr2[i] = num5.intValue();
            i++;
        }
        iArr[i] = new int[0];
        iArr2[i] = this.mTextColor;
        int i2 = i + 1;
        if (i2 != 6) {
            int[][] iArr3 = new int[i2][];
            int[] iArr4 = new int[i2];
            System.arraycopy(iArr, 0, iArr3, 0, i2);
            System.arraycopy(iArr2, 0, iArr4, 0, i2);
            iArr = iArr3;
            iArr2 = iArr4;
        }
        return new ColorStateList(iArr, iArr2);
    }

    public void intoTextColor() {
        if (isTextGradientColor()) {
            TextView textView = this.mTextView;
            textView.setText(buildLinearGradientSpannable(textView.getText()));
            return;
        }
        this.mTextView.setTextColor(buildColorState());
    }
}

File: decompiled/sources/com/hjq/shape/builder/TextColorBuilder.kt
package com.hjq.shape.builder

import android.content.res.ColorStateList
import android.content.res.TypedArray
import android.text.SpannableStringBuilder
import android.widget.TextView
import com.hjq.shape.other.LinearGradientFontSpan
import com.hjq.shape.styleable.ITextColorStyleable

class TextColorBuilder(
    private val mTextView: TextView,
    typedArray: TypedArray,
    iTextColorStyleable: ITextColorStyleable
) {
    private var mTextCheckedColor: Int? = null
    private var mTextColor: Int
    private var mTextDisabledColor: Int? = null
    private var mTextFocusedColor: Int? = null
    private var mTextGradientColor: IntArray? = null
    private var mTextGradientOrientation: Int
    private var mTextPressedColor: Int? = null
    private var mTextSelectedColor: Int? = null

    init {
        mTextColor = typedArray.getColor(
            iTextColorStyleable.getTextColorStyleable(),
            mTextView.textColors.defaultColor
        )
        if (typedArray.hasValue(iTextColorStyleable.getTextPressedColorStyleable())) {
            mTextPressedColor =
                typedArray.getColor(iTextColorStyleable.getTextPressedColorStyleable(), mTextColor)
        }
        if (
            iTextColorStyleable.getTextCheckedColorStyleable() > 0 &&
            typedArray.hasValue(iTextColorStyleable.getTextCheckedColorStyleable())
        ) {
            mTextCheckedColor =
                typedArray.getColor(iTextColorStyleable.getTextCheckedColorStyleable(), mTextColor)
        }
        if (typedArray.hasValue(iTextColorStyleable.getTextDisabledColorStyleable())) {
            mTextDisabledColor =
                typedArray.getColor(iTextColorStyleable.getTextDisabledColorStyleable(), mTextColor)
        }
        if (typedArray.hasValue(iTextColorStyleable.getTextFocusedColorStyleable())) {
            mTextFocusedColor =
                typedArray.getColor(iTextColorStyleable.getTextFocusedColorStyleable(), mTextColor)
        }
        if (typedArray.hasValue(iTextColorStyleable.getTextSelectedColorStyleable())) {
            mTextSelectedColor =
                typedArray.getColor(iTextColorStyleable.getTextSelectedColorStyleable(), mTextColor)
        }
        if (
            typedArray.hasValue(iTextColorStyleable.getTextStartColorStyleable()) &&
            typedArray.hasValue(iTextColorStyleable.getTextEndColorStyleable())
        ) {
            mTextGradientColor =
                if (typedArray.hasValue(iTextColorStyleable.getTextCenterColorStyleable())) {
                    intArrayOf(
                        typedArray.getColor(iTextColorStyleable.getTextStartColorStyleable(), mTextColor),
                        typedArray.getColor(iTextColorStyleable.getTextCenterColorStyleable(), mTextColor),
                        typedArray.getColor(iTextColorStyleable.getTextEndColorStyleable(), mTextColor)
                    )
                } else {
                    intArrayOf(
                        typedArray.getColor(iTextColorStyleable.getTextStartColorStyleable(), mTextColor),
                        typedArray.getColor(iTextColorStyleable.getTextEndColorStyleable(), mTextColor)
                    )
                }
        }
        mTextGradientOrientation =
            typedArray.getColor(iTextColorStyleable.getTextGradientOrientationStyleable(), 0)
    }

    fun setTextColor(color: Int?): TextColorBuilder {
        mTextColor = color!!
        clearTextGradientColor()
        return this
    }

    fun getTextColor(): Int? {
        return mTextColor
    }

    fun setTextPressedColor(color: Int?): TextColorBuilder {
        mTextPressedColor = color
        return this
    }

    fun getTextPressedColor(): Int? {
        return mTextPressedColor
    }

    fun setTextCheckedColor(color: Int?): TextColorBuilder {
        mTextCheckedColor = color
        return this
    }

    fun getTextCheckedColor(): Int? {
        return mTextCheckedColor
    }

    fun setTextDisabledColor(color: Int?): TextColorBuilder {
        mTextDisabledColor = color
        return this
    }

    fun getTextDisabledColor(): Int? {
        return mTextDisabledColor
    }

    fun setTextFocusedColor(color: Int?): TextColorBuilder {
        mTextFocusedColor = color
        return this
    }

    fun getTextFocusedColor(): Int? {
        return mTextFocusedColor
    }

    fun setTextSelectedColor(color: Int?): TextColorBuilder {
        mTextSelectedColor = color
        return this
    }

    fun getTextSelectedColor(): Int? {
        return mTextSelectedColor
    }

    fun setTextGradientColor(color: IntArray?): TextColorBuilder {
        mTextGradientColor = color
        return this
    }

    fun getTextGradientColor(): IntArray? {
        return mTextGradientColor
    }

    fun setTextGradientOrientation(orientation: Int): TextColorBuilder {
        mTextGradientOrientation = orientation
        return this
    }

    fun getTextGradientOrientation(): Int? {
        return mTextGradientOrientation
    }

    val isTextGradientColor: Boolean
        get() {
            val gradientColor = mTextGradientColor
            return gradientColor != null && gradientColor.isNotEmpty()
        }

    fun clearTextGradientColor() {
        mTextGradientColor = null
    }

    fun buildLinearGradientSpannable(charSequence: CharSequence?): SpannableStringBuilder {
        return LinearGradientFontSpan.buildLinearGradientSpannable(
            charSequence!!,
            mTextGradientColor,
            null,
            mTextGradientOrientation
        )
    }

    fun buildColorState(): ColorStateList {
        if (
            mTextPressedColor == null &&
            mTextCheckedColor == null &&
            mTextDisabledColor == null &&
            mTextFocusedColor == null &&
            mTextSelectedColor == null
        ) {
            return ColorStateList.valueOf(mTextColor)
        }
        val states = ArrayList<IntArray>(6)
        val colors = ArrayList<Int>(6)
        mTextPressedColor?.let {
            states.add(intArrayOf(16842919))
            colors.add(it)
        }
        mTextCheckedColor?.let {
            states.add(intArrayOf(16842912))
            colors.add(it)
        }
        mTextDisabledColor?.let {
            states.add(intArrayOf(-16842910))
            colors.add(it)
        }
        mTextFocusedColor?.let {
            states.add(intArrayOf(16842908))
            colors.add(it)
        }
        mTextSelectedColor?.let {
            states.add(intArrayOf(16842913))
            colors.add(it)
        }
        states.add(intArrayOf())
        colors.add(mTextColor)
        return ColorStateList(states.toTypedArray(), colors.toIntArray())
    }

    fun intoTextColor() {
        if (isTextGradientColor) {
            mTextView.text = buildLinearGradientSpannable(mTextView.text)
            return
        }
        mTextView.setTextColor(buildColorState())
    }
}
